exonum/src/events/noise/wrappers/sodium_wrapper/handshake.rs

```diff
@@ -97,15 +97,27 @@ impl NoiseHandshake {
     pub fn finalize<S: AsyncRead + AsyncWrite + 'static>(
         self,
         stream: S,
-    ) -> Result<Framed<S, MessagesCodec>, io::Error> {
+    ) -> Result<(Framed<S, MessagesCodec>, x25519::PublicKey), io::Error> {
         let noise = self.noise.into_transport_mode()?;
+        let remote_static_key = {
```

Comment:
Nit: The separate scope for `rs` seems to be a bit superfluous; something like this would seemingly do fine:

```rust
let remote_static_key = noise
    .session
    .get_remote_static()
    .expect("Remote static key is not present!");
let remote_static_key = x25519::PublicKey::from_slice(remote_static_key)
    .expect("Remote static key is not valid x25519 key!");
```

(Admittedly, I haven't try to compile this code, so maybe I'm missing something.)

Reply:
We cannot borrow `noise` in the same scope, because later we need to move it to the return value.